gen/installer/aws.py

```diff
@@ -138,6 +215,24 @@ def get_test_session(config=None):
     return release.call_matching_arguments(release.storage.aws.get_session, config, True)
 
 
+def gen_ami_mapping(mappings):
```

Comment:
nit: if we made region_to_ami_map an actual map ratherthan a list which looks like:
```
{
  'ap-northeast-1': {
    'coreos': '',
    'stable': '',
    'el7': '',
    'natami': ''
  }
}
```

then we could really easily test the ami mapping wihtout copying the AMI ids around:
```python
assert result["ap-northeast-1"] == {'stable': gen.installer.aws.region_to_ami_map['ap-northeast-1']['stable']}
```

which makes the test less fragile (every time someone updates the AMIs currently they have to touch this test which they'll never think of until CI pings them). It also is a lot simpler than what you have to do with the list thing for the same effect:
```python
ap_northeast_1_list = list(filter(lambda x: x['region'] == 'ap-northeast-1', gen.installer.aws.region_to_ami_map))
assert len(ap_northeast_1_list) == 1
assert result["ap-northeast-1"] == {'stable': ap_northeast_1_list[0]['stable']}
```

It also makes gen_ami_mapping a little simpler:
```python
def gen_ami_mapping(mappings):
    # create new dict with required mappings
    # all will have region by default
    for region, amis in region_to_ami_map.items():
        # Build the final dictionary
        final[region] = dict()
        for key in mappings:
          # TODO(cmaloney): natami needs to be mapped to the `default` key for 
          # now. Should update the templates to remove the need.
          final_key = 'default' if key == 'natami' else key
          final[region][final_key] = amis[key]

    # Returning -1:1 to cut the curly braces off the outside of the json dictionary so it formats
    # right into the templates.
    return json.dumps(final, indent=4, sort_keys=True)[1:-1]
```